File: Modules/ServerTypes/TaskFlow/ProjectColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProjectRow } from "./ProjectRow";

export interface ProjectColumns {
    Name: Column<ProjectRow>;
    WorkspaceName: Column<ProjectRow>;
    Description: Column<ProjectRow>;
    CreatedAt: Column<ProjectRow>;
    UpdatedAt: Column<ProjectRow>;
}

export class ProjectColumns extends ColumnsBase<ProjectRow> {
    static readonly columnsKey = 'TaskFlow.Project';
    static readonly Fields = fieldsProxy<ProjectColumns>();
}

File: Modules/TaskFlow/Project/ProjectColumns.cs
namespace TaskFlowSerenity.TaskFlow.Columns;

[ColumnsScript("TaskFlow.Project")]
[BasedOnRow(typeof(ProjectRow), CheckNames = true)]
public class ProjectColumns
{
    [EditLink]
    public string Name { get; set; }
    public string WorkspaceName { get; set; }
    public string Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}